cg/services/illumina_services/illumina_post_processing_service/housekeeper_storage.py

```diff
@@ -0,0 +1,144 @@
+    get_sample_fastqs_from_flow_cell,
+    get_undetermined_fastqs,
+    rename_fastq_file_if_needed,
+)
+from cg.store.store import Store
+from cg.utils.files import get_files_matching_pattern
+
+LOG = logging.getLogger(__name__)
+
+
+def _check_if_fastq_path_should_be_stored_in_housekeeper(
+    sample_id: str,
+    sample_fastq_path: Path,
+    sequencer_type: Sequencers,
+    device_internal_id: str,
+    store: Store,
+) -> bool:
+    """
+    Check if a sample fastq file should be tracked in Housekeeper.
+    Only fastq files that pass the q30 threshold should be tracked.
+    """
+    lane = get_lane_from_sample_fastq(sample_fastq_path)
+    q30_threshold: int = get_q30_threshold(sequencer_type)
+
+    metric = store.get_illumina_metrics_entry_by_device_sample_and_lane(
+        device_internal_id=device_internal_id,
+        sample_internal_id=sample_id,
+        lane=lane,
+    )
+
+    if metric:
+        return metric.base_passing_q30_percent >= q30_threshold
+
+    LOG.warning(
+        f"Skipping fastq file {sample_fastq_path.name} as no metrics entry was found in status db."
+    )
+    LOG.warning(f"Flow cell name: {device_internal_id}, sample id: {sample_id}, lane: {lane} ")
+    return False
+
+
+def add_sample_fastq_files_to_housekeeper(
+    run_directory_data: IlluminaRunDirectoryData, hk_api: HousekeeperAPI, store: Store
+) -> None:
+    """Add sample fastq files from the demultiplex directory to Housekeeper."""
+    sample_internal_ids: list[str] = run_directory_data.sample_sheet.get_sample_ids()
+    for sample_internal_id in sample_internal_ids:
+        sample_fastq_paths: list[Path] | None = get_sample_fastqs_from_flow_cell(
+            flow_cell_directory=run_directory_data.get_demultiplexed_runs_dir(),
+            sample_internal_id=sample_internal_id,
+        )
+        if not sample_fastq_paths:
+            LOG.warning(
+                f"Cannot find fastq files for sample {sample_internal_id} in {run_directory_data.path}. Skipping."
+            )
+            continue
+        for sample_fastq_path in sample_fastq_paths:
+            sample_fastq_path: Path = rename_fastq_file_if_needed(
+                fastq_file_path=sample_fastq_path, flow_cell_name=run_directory_data.id
+            )
+            if _check_if_fastq_path_should_be_stored_in_housekeeper(
+                sample_id=sample_internal_id,
+                sample_fastq_path=sample_fastq_path,
+                sequencer_type=run_directory_data.sequencer_type,
+                device_internal_id=run_directory_data.id,
+                store=store,
+            ):
+                hk_api.store_fastq_path_in_housekeeper(
+                    sample_internal_id=sample_internal_id,
+                    sample_fastq_path=sample_fastq_path,
+                    flow_cell_id=run_directory_data.id,
+                )
+
+
+def store_undetermined_fastq_files(
+    run_directory_data: IlluminaRunDirectoryData, hk_api: HousekeeperAPI, store: Store
```

Comment:
or "demux_run_dir_data" / "seq_run_dir_data"

Reply:
To remove the ambiguity, I will use methods  `get_sequencing_runs_dir` or `get_demultiplexed_runs_dir` instead of attribute `path` in an according way